src/libs/elektra/keyset.c

```diff
@@ -1560,7 +1560,7 @@ Key * ksAtCursor (KeySet * ks, elektraCursor pos)
 {
 	if (!ks) return 0;
 	if (pos < 0) return 0;
-	if (ks->size < (size_t) pos) return 0;
+	if (ks->size <= (size_t) pos) return 0;
```

Comment:
This looks like a major bug, did you also add a test case?

Reply:
It doesn't actually change anything, because AFAIK the way `KeySet`s are allocated means that, if `ks->size == 0` then `ks->array[pos] == 0`, so the return value doesn't change.

Reply:
The edge case here is when `ks->size == pos`, which would be the last element of the keyset array. The last element is always `0`, regardless of `ks->size`, which is why the bug never took effect.

I haven't tested it but the change looks correct to me as well.